src/app/blog/page.tsx

```diff
@@ -1,70 +1,36 @@
-import dynamic from 'next/dynamic'
+import type { AsyncNextServerSearchParams } from '@/types/searchParams'
 
-import { BookOpen } from '@phosphor-icons/react/dist/ssr'
-
-import { type NextServerSearchParams } from '@/types/searchParams'
-
-import { DEFAULT_CATEGORY_FILTER_OPTION } from '@/constants/filterConstants'
 import { PATHS } from '@/constants/paths'
-import { CATEGORY_KEY } from '@/constants/searchParams'
 
 import { graphicsData } from '@/data/graphicsData'
 
-import { buildImageSizeProp } from '@/utils/buildImageSizeProp'
-import { getCategoryLabel } from '@/utils/categoryUtils'
 import { createMetadata } from '@/utils/createMetadata'
 import { extractSlugFromFilename } from '@/utils/fileUtils'
-import { entryMatchesCategoryQuery } from '@/utils/filterUtils'
 import { getFrontmatter } from '@/utils/getFrontmatter'
 import { getSortOptions } from '@/utils/getSortOptions'
-import { normalizeQueryParam } from '@/utils/queryUtils'
 
 import { FeaturedPageFrontmatterSchema } from '@/schemas/FrontmatterSchema'
 
-import { useFilter } from '@/hooks/useFilter'
-import { useListboxOptions } from '@/hooks/useListboxOptions'
-import { usePagination } from '@/hooks/usePagination'
-import { useSearch } from '@/hooks/useSearch'
-import { useSort } from '@/hooks/useSort'
-
-import { Card } from '@/components/Card'
-import { CardGrid } from '@/components/CardGrid'
-import { CategoryFilter } from '@/components/CategoryFilter'
-import { FilterContainer } from '@/components/FilterContainer'
-import { NoSearchResultsMessage } from '@/components/NoSearchResultsMessage'
 import { PageHeader } from '@/components/PageHeader'
 import { PageLayout } from '@/components/PageLayout'
 import { PageSection } from '@/components/PageSection'
-import { Search } from '@/components/Search'
-import { Sort } from '@/components/Sort'
 import { StructuredDataScript } from '@/components/StructuredDataScript'
 
+import { BlogContent } from './components/BlogContent'
 import { blogSortConfigs } from './constants/sortConfigs'
 import { generateStructuredData } from './utils/generateStructuredData'
 import { getBlogPostData, getBlogPostsData } from './utils/getBlogPostData'
 import { getMetaData } from './utils/getMetaData'
 
-const NoSSRPagination = dynamic(
-  () => import('@/components/Pagination').then((module) => module.Pagination),
-  { ssr: false },
-)
-
 type Props = {
-  searchParams: NextServerSearchParams
+  searchParams: Promise<AsyncNextServerSearchParams>
```

Comment:
`AsyncNextServerSearchParams` already includes a Promise no?


```suggestion
  searchParams: AsyncNextServerSearchParams
```